<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1345</width>
    <height>829</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPlainTextEdit" name="input_editor">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>281</width>
      <height>301</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="input_label">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>10</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>输入文法</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="submit_button">
    <property name="geometry">
     <rect>
      <x>311</x>
      <y>529</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>LL(1)分析过程</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="output_browser">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>30</y>
      <width>791</width>
      <height>751</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QLabel" name="output_label">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>10</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>输出</string>
    </property>
   </widget>
   <widget class="QPushButton" name="submit_button_2">
    <property name="geometry">
     <rect>
      <x>311</x>
      <y>569</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>LR(0)分析过程</string>
    </property>
   </widget>
   <widget class="QPushButton" name="submit_button_3">
    <property name="geometry">
     <rect>
      <x>311</x>
      <y>609</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>SLR(1)分析过程</string>
    </property>
   </widget>
   <widget class="QPushButton" name="show_output_button">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>750</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>查看完整输出文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="get_set_button">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>410</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>75</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="text">
     <string>生成集合</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="first_set_browser">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>131</width>
      <height>331</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="follow_set_browser">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>450</y>
      <width>141</width>
      <height>331</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="run_all_button">
    <property name="geometry">
     <rect>
      <x>309</x>
      <y>120</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>&gt;&gt;&gt;分析文法&gt;&gt;&gt;&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>150</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>判断是哪种文法并输出对应分析表</string>
    </property>
    <property name="scaledContents">
     <bool>false</bool>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>420</y>
      <width>36</width>
      <height>8</height>
     </rect>
    </property>
    <property name="text">
     <string>↓↓↓</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>420</y>
      <width>36</width>
      <height>8</height>
     </rect>
    </property>
    <property name="text">
     <string>↓↓↓</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>10</y>
      <width>121</width>
      <height>101</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;输入要求：首行是行数，非终结符用大写字母表示，终结符用小写字母表示，ε用~表示；当产生式有多个候选式时应分为多个产生式；拓广文法自行输入&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="load_sample_button">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>249</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>填入LL(1)样例2</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load_sample_button_2">
    <property name="geometry">
     <rect>
      <x>309</x>
      <y>289</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>填入LR(0)样例1</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load_sample_button_3">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>369</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>填入SLR(1)样例</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load_sample_button_4">
    <property name="geometry">
     <rect>
      <x>309</x>
      <y>209</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>填入LL(1)样例1</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load_sample_button_5">
    <property name="geometry">
     <rect>
      <x>309</x>
      <y>329</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>填入LR(0)样例2</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load_sample_button_6">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>489</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>填入其他类样例</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>189</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>301</x>
      <y>509</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>730</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>409</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>填入LR(1)样例</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>449</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>填入LALR(1)样例</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>311</x>
      <y>649</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>LR(1)分析过程</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>311</x>
      <y>689</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>LALR(1)分析过程</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>360</y>
      <width>281</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>340</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>输入语句</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_fa">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>1250</x>
      <y>150</y>
      <width>16</width>
      <height>191</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="testButton">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>590</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>绘制FA</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1345</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
